toxic_timeout.go

```diff
@@ -23,7 +29,22 @@ func (t *TimeoutToxic) SetEnabled(enabled bool) {
 }
 
 func (t *TimeoutToxic) Pipe(stub *ToxicStub) {
+
+
 	timeout := time.Duration(t.Timeout) * time.Millisecond
+
+	// Percent of messages that will be passed without any effect of this timeout feature.
+	// When not set, skip = 0 by default, all messages are affected as usual.
+	skip := int32(t.Skip)	
+		
+	// Do not apply this timeout to this percent of packages.
+	if(skip>0){
+		var randValue = rand.Int31n(100);
+		if((randValue+1) <= skip){
+			return
```

Comment:
A toxic should only really return if it has received an interrupt or it is closing the connection. This will just cause the link to freeze.
You will need to pipe data through yourself like in the noop toxic if that's the intention here.

I also think in this case the `toxicity` field in 2.0 is exactly what you're looking to do here.